accounting_partner_category/models/account_move.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 Foodles (https://www.foodles.com/)
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _name = "account.move"
+    _inherit = "account.move"
+
+    category_ids = fields.Many2many(
+        "accounting.partner.category",
+        string="Tags",
+        help="Tags when partner was added to the invoice.",
+    )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        super()._onchange_partner_id()
+        self.category_ids = (
+            self.partner_id.commercial_partner_id.accounting_category_ids
+        )
```

Comment:
Hi, I think you should use a computed field. Onchange won't be triggered when importing invoices from other sources (or just if created anywhere else than in the form view).
You can take a look how it was done there
https://github.com/OCA/account-invoicing/blob/15.0/account_invoice_transmit_method/models/account_move.py

Reply:
the problem with such api.depends implementation, when creating new invoice and you want to remove all tags they will appear while saving as api.depends is called after create where partner_id will be considered changed so the user input will be lost given some frustrations

Reply:
@jguenat I'm not sure how to avoid this side effect ?

So I've also implement #1302 when creating invoice from sale order at the moment...

Also if invoice are imported from other source there is a chance that they already call [upstream `_onchange_partner_id` method](https://github.com/OCA/OCB/blob/14.0/addons/account/models/account_move.py#L436) to get proper informations on invoice.

Reply:
If you have the compute="_compute_your field" and readonly=False defined in conjunction with the @api.depends I'm pretty sure the behaviour is similar to an onchange on the form view with the benefit of the compute called on creation in any cases. You can check the functionality of account_invoice_transmit_method :)

Reply:
My bad, I probably missed something during my initial implementation/test... I'll close #1302 then :)